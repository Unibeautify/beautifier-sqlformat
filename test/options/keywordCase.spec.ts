import * as fs from "fs";
import * as path from "path";

import { newUnibeautify, Beautifier } from "unibeautify";
import beautifier from "../../src";
import { raw } from "../utils";

// tslint:disable:mocha-no-side-effect-code
test(`should uppercase SQL keywords`, () => {
  const text: string = fs
    .readFileSync(path.resolve(__dirname, `../fixtures/test2.sql`))
    .toString();
  const unibeautify = newUnibeautify();
  unibeautify.loadBeautifier(beautifier);
  return unibeautify
    .beautify({
      languageName: "SQL",
      options: {
        SQL: {
          keyword_case: "uppercase",
        },
      },
      text,
    })
    .then(results => {
      expect(raw(results)).toMatchSnapshot();
    });
});
test(`should capitalize SQL keywords`, () => {
  const text: string = fs
    .readFileSync(path.resolve(__dirname, `../fixtures/test2.sql`))
    .toString();
  const unibeautify = newUnibeautify();
  unibeautify.loadBeautifier(beautifier);
  return unibeautify
    .beautify({
      languageName: "SQL",
      options: {
        SQL: {
          keyword_case: "capitalize",
        },
      },
      text,
    })
    .then(results => {
      expect(raw(results)).toMatchSnapshot();
    });
});
test(`should lowercase SQL keywords`, () => {
  const text: string = fs
    .readFileSync(path.resolve(__dirname, `../fixtures/test2.sql`))
    .toString();
  const unibeautify = newUnibeautify();
  unibeautify.loadBeautifier(beautifier);
  return unibeautify
    .beautify({
      languageName: "SQL",
      options: {
        SQL: {
          keyword_case: "lowercase",
        },
      },
      text,
    })
    .then(results => {
      expect(raw(results)).toMatchSnapshot();
    });
});
