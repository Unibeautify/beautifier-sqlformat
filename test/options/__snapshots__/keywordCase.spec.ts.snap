// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should capitalize SQL keywords 1`] = `
Object {
  Symbol(raw): "Select *
From Users u
Join users_permission up On u.user_id = up.user_id
Where u.user_id = 1
  And up.permission_id = 2;",
}
`;

exports[`should lowercase SQL keywords 1`] = `
Object {
  Symbol(raw): "select *
from Users u
join users_permission up on u.user_id = up.user_id
where u.user_id = 1
  and up.permission_id = 2;",
}
`;

exports[`should uppercase SQL keywords 1`] = `
Object {
  Symbol(raw): "SELECT *
FROM Users u
JOIN users_permission up ON u.user_id = up.user_id
WHERE u.user_id = 1
  AND up.permission_id = 2;",
}
`;
